src/core/geometry/qgsgeometry.h

```diff
@@ -1354,6 +1394,81 @@ class CORE_EXPORT QgsGeometry
      */
     QgsGeometry poleOfInaccessibility( double precision, double *distanceToBoundary SIP_OUT = nullptr ) const;
 
+    /**
+     * Constructs the Largest Empty Circle for a set of obstacle geometries, up to a
+     * specified tolerance.
+     *
+     * The Largest Empty Circle is the largest circle which has its center in the convex hull of the
+     * obstacles (the boundary), and whose interior does not intersect with any obstacle.
+     * The circle center is the point in the interior of the boundary which has the farthest distance from
+     * the obstacles (up to tolerance). The circle is determined by the center point and a point lying on an
+     * obstacle indicating the circle radius.
+     * The implementation uses a successive-approximation technique over a grid of square cells covering the obstacles and boundary.
+     * The grid is refined using a branch-and-bound algorithm.  Point containment and distance are computed in a performant
+     * way by using spatial indexes.
+     * Returns a two-point linestring, with one point at the center of the inscribed circle and the other
+     * on the boundary of the inscribed circle.
+     *
+     * This method requires QGIS builds based on GEOS 3.9 or later.
+     *
+     * \throws QgsNotSupportedException on QGIS builds based on GEOS 3.8 or earlier.
+     *
+     * \since QGIS 3.20
+     */
+    QgsGeometry largestEmptyCircle( double tolerance, const QgsGeometry &boundary = QgsGeometry() ) const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns a linestring geometry which represents the minimum diameter of the geometry.
+     *
+     * The minimum diameter is defined to be the width of the smallest band that
+     * contains the geometry, where a band is a strip of the plane defined
+     * by two parallel lines. This can be thought of as the smallest hole that the geometry
+     * can be moved through, with a single rotation.
+     *
+     * This method requires QGIS builds based on GEOS 3.6 or later.
+     *
+     * \throws QgsNotSupportedException on QGIS builds based on GEOS 3.5 or earlier.
+     *
+     * \since QGIS 3.20
+     */
+    QgsGeometry minimumWidth() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Computes the minimum clearance of a geometry.
+     *
+     * The minimum clearance is the smallest amount by which
+     * a vertex could be move to produce an invalid polygon, a non-simple linestring, or a multipoint with
+     * repeated points.  If a geometry has a minimum clearance of 'eps', it can be said that:
+     *
+     * - No two distinct vertices in the geometry are separated by less than 'eps'
+     * - No vertex is closer than 'eps' to a line segment of which it is not an endpoint.
+     *
+     * If the minimum clearance cannot be defined for a geometry (such as with a single point, or a multipoint
+     * whose points are identical, a value of Infinity will be calculated.
```

Comment:
I think that a `)` is missing somewhere.

Reply:
Fixed